crates/rpc/rpc/src/otterscan.rs

```diff
@@ -198,10 +198,62 @@ where
     /// Handler for `getTransactionBySenderAndNonce`
     async fn get_transaction_by_sender_and_nonce(
         &self,
-        _sender: Address,
-        _nonce: u64,
-    ) -> RpcResult<Option<Transaction>> {
-        Err(internal_rpc_err("unimplemented"))
+        sender: Address,
+        nonce: u64,
+    ) -> RpcResult<Option<TxHash>> {
+        // Check if the sender is a contract
+        if self.has_code(sender, None).await? {
+            return Ok(None)
+        }
+
+        let highest =
+            self.eth.transaction_count(sender, None).await?.saturating_to::<u64>();
+
+        // If the nonce is higher or equal to the highest nonce, the transaction is pending or not
+        // exists.
+        if nonce >= highest {
+            return Ok(None)
+        }
+
+        // To find the first block where the sender's nonce is equal to the given one, we can use
+        // binary search within the range of blocks starting from block 1 up to the latest block
+        // number.
+        let mut low = 1;
+        let mut high = self.eth.block_number()?.saturating_to::<u64>();
+        let mut num = high;
+
+        while low <= high {
+            let mid = (low + high) / 2;
+            let mid_nonce = EthApiServer::transaction_count(
+                &self.eth,
+                sender,
+                Some(BlockId::Number(BlockNumberOrTag::Number(mid))),
+            )
+            .await?
+            .saturating_to::<u64>();
+
+            // The `transaction_count` returns the `nonce` after the transaction was executed, which
+            // is the state of the account after the block, and we need to find the
+            // transaction whose nonce is the pre-state, so need to compare with
+            // `nonce+1`.
+            if mid_nonce < nonce + 1 {
+                low = mid + 1; // not found in current block, need to search in the later blocks
+            } else {
+                high = mid - 1; // found in current block, try to find a lower block
+                num = mid;
+            }
+        }
+
+        let Some(BlockTransactions::Full(transactions)) =
+            self.eth.block_by_number(num.into(), true).await?.map(|block| block.inner.transactions)
```

Comment:
```suggestion
            self.eth.block_by_number((num - 1).into(), true).await?.map(|block| block.inner.transactions)
```

Reply:
I think the change is not correct, if found nonce in block N, need to get txs in block N, not N-1

Reply:
Erigon is searching nonce from the `AccountsHistory` table by tx index, but we are searching in the block body instead, so the implementation is different. 

eg: if we found the nonce changed in block 10, then iterate the txs in block 9 can't get anything.

Reply:
if you added some test to show this, I would be strongly convinced! you can compare without output from region tests for same inputs